src/main/java/utils/ApplicationProgrammingInterface.java

```diff
@@ -16,7 +16,9 @@
 import java.util.LinkedHashSet;
 import java.util.Set;
 import java.util.logging.Logger;
-
+/**
+ * SOLID: DIP vi ApplicationProgrammingInterface cai dat truc tiep cac chuc nang ma ko qua interface hay abstract
+ * */
 public class ApplicationProgrammingInterface {
 
 	public static DateFormat DATE_FORMATTER = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
```

Comment:
I detect that this code is problematic. According to the [Multithreaded correctness (MT_CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#multithreaded-correctness-mt-correctness), [STCAL: Static DateFormat (STCAL_STATIC_SIMPLE_DATE_FORMAT_INSTANCE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#stcal-static-dateformat-stcal-static-simple-date-format-instance).
As the JavaDoc states, DateFormats are inherently unsafe for multithreaded use. Sharing a single instance across thread boundaries without proper synchronization will result in erratic behavior of the application.
You may also experience serialization problems.
Using an instance field is recommended.
For more information on this see [JDK Bug #6231579](http://bugs.java.com/bugdatabase/view_bug.do?bug_id=6231579) and [JDK Bug #6178997](http://bugs.java.com/bugdatabase/view_bug.do?bug_id=6178997).